frontend/src/utilities/hooks/use-identity.ts

```diff
@@ -0,0 +1,64 @@
+import IdentityRecord from "models/view-models/identity-record";
+import { useCallback } from "react";
+import UserLoginRecord from "models/view-models/user-login-record";
+import RoleService from "utilities/services/role-service";
+import UserService from "utilities/services/user-service";
+import UserRecord from "models/view-models/user-record";
+import RoleRecord from "models/view-models/role-record";
+
+/**
+ * Custom hook providing utility functions for the current identity of the user
+ */
+export default function useIdentity() {
```

Comment:
Did a fairly significant refactor on this hook to make it more manageable. There are more opportunities once we get some of the ts types for service hooks done https://github.com/AndcultureCode/AndcultureCode.JavaScript.React/issues/18

Reply:
The idea is that 'getRole' and 'getUser' could be external to the `useIdentity` function all together.

Reply:
Now broken out into smaller functions will be easier to do async api calls instead of majorly synchronous blocking loading of the original version.